lib/get-config.js

```diff
@@ -303,7 +307,56 @@ function validateRules(config, options) {
   }
 }
 
-module.exports = function(options) {
+function _determineConfigForSeverity(config) {
+  switch (config) {
+    case 'off':
+      return { config: false, severity: IGNORE_SEVERITY };
+    case 'warn':
+      return { config: true, severity: WARNING_SEVERITY };
+    case 'error':
+      return { config: true, severity: ERROR_SEVERITY };
+  }
+}
+
+/**
+ * Sets the severity and config on the rule object.
+ * Eg:
+ * {'no-implicit-this': 'warn'} -> {'no-implicit-this': { severity: 'warn', config: true }}
+ * { 'no-implicit-this': [ 'warn', { allow: [ 'fooData' ] } ] }
+ * would become:
+ * { 'no-implicit-this': { severity: 'warn', config: 'allow': [ 'fooData' ] } }
+ * @param {*} configData
+ */
+function processRules(config) {
+  for (let key in config.rules) {
+    let severityConfig;
+    let ruleData = config.rules[key];
+    // defaults to error always.
+    let ruleConfig = {
+      severity: ruleData === false ? IGNORE_SEVERITY : ERROR_SEVERITY,
+      config: ruleData,
+    };
+    // In case of {'no-implicit-this': 'off|warn|error'}
+    if (typeof ruleData === 'string') {
+      severityConfig = _determineConfigForSeverity(ruleData);
+      if (severityConfig) {
+        ruleConfig.severity = severityConfig.severity;
+        ruleConfig.config = severityConfig.config;
+      }
+    } else if (Array.isArray(ruleData)) {
+      // array of severity and custom rule config
+      let severity = ruleData[0];
+      severityConfig = _determineConfigForSeverity(severity);
+      if (severityConfig) {
+        ruleConfig.severity = severityConfig.severity;
+        ruleConfig.config = ruleData[1] !== undefined ? ruleData[1] : severityConfig.config;
```

Comment:
What is this ternary for? I think we should just use `ruleData[1]` here, regardless of what it is (when using the severity first syntax, the second array argument is **completely** opaque to us. It can be any value...).

Reply:
the reason being for cases where:
`{rules: {'foo': ['warn']}}`
But since now we are checking for `severity` to bypass building rules, I think its safe to remove

Reply:
👍, please remove (so we can make no assumptions about that `rulesData[1]` value)